### PR TITLE
Add CSI Plug-N-Play Design

### PR DESCRIPTION
**Pull request by asifdxtreme**
**What type of PR is this?**

/kind new feature

**What this PR does / why we need it**:
This PR adds the design of new CSI Plug-N-Play mechanism

### CODE REVIEW

**Review by kumarashit — APPROVED** [review]
LGTM

**Review by skdwriting — COMMENTED** [review]
1) Please add the current SODA NBP for CSI and difference aspects
2) Also clearly mention the issues on the API flow (horizontally to SODA)
3) Also mention, we will have the current CSI and this design for exploration and testing and finalize one design which is better. So the refactoring is expected

**Comment by asifdxtreme** [comment]
> 1. Please add the current SODA NBP for CSI and difference aspects
> 2. Also clearly mention the issues on the API flow (horizontally to SODA)
> 3. Also mention, we will have the current CSI and this design for exploration and testing and finalize one design which is better. So the refactoring is expected

@skdwriting Done. Added all the details in the design docs

**Review by skdwriting — CHANGES_REQUESTED** [review]
Can we update the existing design document and add the new approach as an alternate design.
Then we provide pros/cons and conclude accordingly.
Otherwise, it will create duplicate design docs for CSI plug and play

**Comment by asifdxtreme** [comment]
> Can we update the existing design document and add the new approach as an alternate design.
> Then we provide pros/cons and conclude accordingly.
> Otherwise, it will create duplicate design docs for CSI plug and play

@skdwriting I have updated the PR with actual design and an alternate design , [preview :](https://github.com/asifdxtreme/design-specs/tree/master/specs/csi-plug-n-play)

File: specs/csi-plug-n-play/Readme.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# SODA-CSI Plug-N-Play
```
Comment (skdwriting) [inline specs/csi-plug-n-play/Readme.md:1]: Original file history lost.
Can we make the design document as a design document name?
Also the alternate, if you want a separate file, please name it accordingly and link. Need to cross link both doc.
Else, update the original and avoid the duplicate information. (intro, goals etc...)

**Comment by skdwriting** [comment]
@asifdxtreme as we discussed please raise a new PR

**Pull request closed without merging**